common/namespace/handler.go

```diff
@@ -787,3 +781,65 @@ func (d *HandlerImpl) validateVisibilityArchivalURI(URIString string) error {
 
 	return archiver.ValidateURI(URI)
 }
+
+// validateRetentionDuration ensures that retention duration can't be set below a sane minimum.
```

Comment:
is there really such a thing as a sane minimum for retention?  What if "until completion" is good enough for a client, and they don't need *any* retention.  In CDS we can handle such a Workflow without *ever* persisting it to disk.  That's a huge win for throughput.  While I agree "that's insane!" :-) I'd hate to see you make that impossible if that's what the customer needs.